srml/treasury/src/lib.rs

```diff
@@ -14,7 +14,48 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! The Treasury: Keeps account of the taxed cash and handles its deployment.
+//! # Treasury Module
+//! 
+//! The `treasury` module keeps account of currency in a `pot` and manages the subsequent
+//! deployment of these funds.
+//! 
+//! ## Overview
+//! 
+//! Funds for treasury are raised in two ways:
+//! 1. By minting new tokens, leading to inflation, and
+//! 2. By channelling some of the tokens set for burning from transaction fees and slashing.
```

Comment:
Doesn't matter what they were meant for, and other modules may change what they are meant for.

```suggestion
//! 2. By channeling tokens from transaction fees and slashing.
```